lib/block_view_message.go

```diff
@@ -142,6 +78,9 @@ func (bav *UtxoView) GetMessagingGroupKeyToMessagingGroupEntryMapping(
 
 	// If an entry exists in the in-memory map, return the value of that mapping.
 	if mapValue, exists := bav.MessagingGroupKeyToMessagingGroupEntry[*messagingGroupKey]; exists {
+		if mapValue.isDeleted {
+			return nil
+		}
```

Comment:
I think we actually want to return the mapValue if it's deleted in case one of the downstream functions is going to merge results from badger. Should we expose an IsDeleted function on the MessagingGroupEntry and expect downstream functions to properly handle?

Reply:
```suggestion
```
let's delete so this behaves exactly as this function does currently in production